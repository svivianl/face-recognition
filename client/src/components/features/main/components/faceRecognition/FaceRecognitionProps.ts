export interface Box {
  topRow: number;
  rightCol: number;
  bottomRow: number;
  leftCol: number;
}

interface FaceRecognitionProps {
  imageUrl: string;
  // faces?: Box;
  faces?: Box[];
}

export default FaceRecognitionProps;
