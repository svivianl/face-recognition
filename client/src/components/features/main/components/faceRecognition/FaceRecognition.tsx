import React from "react";
import PropTypes from "prop-types";
import FaceRecognitionProps from "./FaceRecognitionProps";
import "../../../../../css/features/main/components/faceRecognition/faceRecognition.css";

const FaceRecognition = ({ imageUrl, faces }: FaceRecognitionProps) => {
  return (
    <div className="d-flex justify-content-center mt-3 mb-3">
      <div className="image-container position-relative">
        <img id="inputimage" alt="" src={imageUrl} />
        {faces &&
          faces.map((box, index) => (
            <div
              key={index}
              className="bounding-box"
              style={{
                top: `${box && box.topRow}%`,
                right: `${box && box.rightCol}%`,
                bottom: `${box && box.bottomRow}%`,
                left: `${box && box.leftCol}%`,
              }}
            ></div>
          ))}
      </div>
    </div>
  );
};

FaceRecognition.propTypes = {
  showIf: PropTypes.bool.isRequired,
  imageUrl: PropTypes.string,
  faces: PropTypes.arrayOf(
    PropTypes.shape({
      topRow: PropTypes.number.isRequired,
      rightCol: PropTypes.number.isRequired,
      bottomRow: PropTypes.number.isRequired,
      leftCol: PropTypes.number.isRequired,
    })
  ),
};

export default FaceRecognition;
