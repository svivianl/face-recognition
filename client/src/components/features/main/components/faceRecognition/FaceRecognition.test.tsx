import React from "react";
import { shallow } from "enzyme";
import FaceRecognition from "./FaceRecognition";
import { Box } from "./FaceRecognitionProps";

describe("FaceRecognition component", () => {
  it("should render without the faces' regions", () => {
    const mockProps = {
      imageUrl: "",
    };
    expect(shallow(<FaceRecognition {...mockProps} />)).toMatchSnapshot();
  });

  it("should render with the faces' regions", () => {
    const faces: Box[] = [
      {
        leftCol: 21.245633,
        topRow: 30.901453,
        rightCol: 69.589236,
        bottomRow: 52.244523,
      },
      {
        leftCol: 68.25825599999999,
        topRow: 21.136338,
        rightCol: 25.463873000000003,
        bottomRow: 64.133796,
      },
      {
        leftCol: 77.968097,
        topRow: 41.064595999999995,
        rightCol: 14.986973999999996,
        bottomRow: 40.807223,
      },
    ];

    const mockProps = {
      imageUrl: "https://samples.clarifai.com/face-det.jpg",
      faces,
    };

    expect(shallow(<FaceRecognition {...mockProps} />)).toMatchSnapshot();
  });
});
