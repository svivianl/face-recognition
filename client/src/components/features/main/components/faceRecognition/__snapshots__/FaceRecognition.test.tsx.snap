// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceRecognition component should render with the faces' regions 1`] = `ShallowWrapper {}`;

exports[`FaceRecognition component should render without the faces' regions 1`] = `ShallowWrapper {}`;
